flytekit/remote/remote.py

```diff
@@ -828,6 +828,44 @@ def register_workflow(
         fwf._python_interface = entity.python_interface
         return fwf
 
+    def register_workflow_script_mode(
+        self,
+        entity: PythonFunctionWorkflow,
+        image_config: typing.Optional[ImageConfig] = None,
+        project: typing.Optional[str] = None,
```

Comment:
I like the idea of using `register_script` for fast register here, but could we add a flag (fast) to `register_workflow` instead. like

```python
def register_workflow(
    self,
    entity: WorkflowBase,
    serialization_settings: typing.Optional[SerializationSettings] = None,
    version: typing.Optional[str] = None,
    fast: bool = False,
    default_launch_plan: typing.Optional[bool] = True,
    options: typing.Optional[Options] = None,
):
   if fast:
       ...
       return self.register_script(...)
   else:
       ...
       return ...
```